holoviews/plotting/bokeh/callbacks.py

```diff
@@ -195,23 +223,38 @@ def set_customjs(self, handle):
         self_callback = self.js_callback.format(comms_target=self.comm.target,
                                                 timeout=self.timeout,
                                                 debounce=self.debounce)
-        attributes = attributes_js(self.attributes)
-        code = 'var data = {};\n' + attributes + self.code + self_callback
 
         handles = {}
         subplots = list(self.plot.subplots.values())[::-1] if self.plot.subplots else []
         plots = [self.plot] + subplots
         for plot in plots:
             handles.update({k: v for k, v in plot.handles.items()
                             if k in self.handles})
+
+        attributes = attributes_js(self.attributes, handles)
+        code = 'var data = {};\n' + attributes + self.code + self_callback
+
+        # Gather the ids of the plotting handles attached to this callback
+        # This allows checking that a stream is not given the state
+        # of a plotting handle it wasn't attached to
+        stream_handle_ids = defaultdict(list)
```

Comment:
Might be good to turn this bit of code into a method (``get_handle_ids``?) and turn the comment into a docstring.